### ISSUE TITLE
[Feature Request] Find seeds with biomes at specific coordinates

### ISSUE THREAD

**Issue by Badel2** [Cubitect/cubiomes#16]
Hi, I think it would be really nice to be able to specify where we want the biomes. My use case is finding the seed of an existing world without having access to it (for example a server).

I suggest to load the biome info from a file with the following format, one condition per line:

```
// X Z BIOME_ID
0 0 1
100 200 7
```

Which would then search for seeds with plains at (0, 0) and river at (100, 200).

Anyway, nice project, keep working on it!

**Comment by Cubitect** [comment]
Currently the my main focus for the repo is to provide a library that allows you to write specific finders like that.  But I might make a small example program with a functionality like that at some point. Thanks for the suggestion.

**Comment by gsquaredxc** [comment]
@Badel2 ok, so I have a working version, but its SUPER slow and written in C++. I'm going to try for GPU acceleration soon.

**Comment by Badel2** [comment]
@gsquaredxc Nice! I was thinking about an efficient way to do the search, because if we need to get the biome at N points there are two options: generate all the area between them (1 call to `genArea` with a big area), or generate one 1x1 area for each point (N calls to `genArea` with 1x1 area). The first approach looks better when the points are close to each other, and the "N calls" approach will probably be slower because everything needs to be calculated again from layer 0 (assuming there is no cache, adding a cache  to some layers would make this approach much faster).

Regarding GPU acceleration, it would be a nice addition but I don't know if it's simple to implement, as the biome generation code is quite large.

**Comment by gogo2464** [comment]
> Currently the my main focus for the repo is to provide a library that allows you to write specific finders like that. But I might make a small example program with a functionality like that at some point. Thanks for the suggestion.

yes! it is the unix philosophy: one stuff does one thing. It is perfect!

We could still use the library as a dependency of a second one to fill the feature of @Badel2 .
